lib/spack/spack/build_systems/cmake.py

```diff
@@ -328,6 +328,12 @@ def flags_to_build_system_args(self, flags):
                 self.cmake_flag_args.append(libs_string.format(lang,
                                                                libs_flags))
 
+    def extra_cmake_args(self):
+        cmake_flags = self.spec.variants.get("cmakeflags", None)
+        if not cmake_flags:
+            return []
+        return [entry for entry in cmake_flags.value.split(' ')]
```

Comment:
I think it'd be better to stick to a multi-valued variant so you get `cmake_flags.value` directly as a tuple and don't have to split on whitespace.

That means flags are passed as `cmake_flags=-DFLAG_A=X,-DFLAG_B=Y`, similar to how flags are passed from compiler to linker: `gcc ... -Wl,-x,-y,-z`.

Reply:
That's a good point, thanks for the suggestion! I just switched to it.

A note: while I was implementing the change, I realized that I had to do it slightly different from `patches` (which is similar from the property management point of view).

https://github.com/spack/spack/blob/5300cbbb2e569190015c72d0970d25425ea38647/lib/spack/spack/solver/asp.py#L1831-L1836

It seems that when `patches` go through the solver, just one value is passed. For `cmakeflags`, instead, I pass all of them, so I can extract and use them later in the build system.

https://github.com/spack/spack/blob/566e3779262588cad3be77230b6bfae76154412c/lib/spack/spack/solver/asp.py#L1841-L1849

I don't know exactly how `patches` works, so I don't know if there's something behind the scenes that I'm missing.